compression_algorithms/huffmann/huffmann.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+    "container/heap"
+    "fmt"
+    "sort"
+)
+
+type Node struct {
+    char   rune
+    freq   int
+    left   *Node
+    right  *Node
+    isLeaf bool
+}
+
+type PriorityQueue []*Node
+
+func (pq PriorityQueue) Len() int { return len(pq) }
+func (pq PriorityQueue) Less(i, j int) bool {
+    if pq[i].freq == pq[j].freq {
+        return pq[i].char < pq[j].char
+    }
+    return pq[i].freq < pq[j].freq
+}
+func (pq PriorityQueue) Swap(i, j int) {
+    pq[i], pq[j] = pq[j], pq[i]
+}
+func (pq *PriorityQueue) Push(x interface{}) {
+    *pq = append(*pq, x.(*Node))
+}
+func (pq *PriorityQueue) Pop() interface{} {
+    old := *pq
+    n := len(old)
+    x := old[n-1]
+    *pq = old[0 : n-1]
+    return x
+}
+
+func BuildHuffmanTree(freqMap map[rune]int) *Node {
```

Comment:
1. Edge Case: Single unique character ( like: "zzzz") not handled
2. Edge Case: Empty character "" not handled

Reply:
![WhatsApp Image 2024-12-23 at 10 53 00_dad30638](https://github.com/user-attachments/assets/41f212fc-fc8f-41a6-8b52-7460dc9df5b9)
The above mentioned cases have been handled

Reply:
Cool then